packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -389,20 +418,46 @@ export class Key extends KeyBase {
     }
   }
 
-  private allowAccountIdentitiesToControl() {
+  /**
+   * Adds the specified principal to the key policy as a key administrator.
+   *
+   * Key administrators have permissions to manage the key (e.g., change permissions, revoke), but do not have permissions
+   * to use the key in cryptographic operations (e.g., encrypt, decrypt).
+   */
+  public addAdmin(principal: iam.IPrincipal) {
```

Comment:
See this thing should have been a `grantAdmin()` in my book.

Or potentially `grantKeyAdministration()`.

Reply:
I agree, and that's where I started out.

Here's what held me back: I *really* wanted -- perhaps without clear enough motivation -- to have the key policy be modified in this case, to match the console experience. I think having the key admins listed clearly on the key policy (as opposed to in the various IAM policies) is a nice bonus for KMS.

However, the "default" implementation via `grant(grantee, perms.ADMIN_ACTIONS)` would actually add the changes to the principal/grantee, not the key policy. Looking a bit closer, I see that there's a `Grant.addToPrincipal`, `Grant.addToPrincipalAndResource`, and `Grant.addToPrincipalOrResource`, all of which add the permissions to the principal (and then optionally also to the resource). There's no way to create a `iam.Grant` to return if you just modify a resource policy. I could add `Grant.addToResource`, but then stopped because I figured there was a good reason it hadn't (yet) been created.

So option A is to stop clinging to the desire to make this match the console experience and just add the admin policies to the principals; option B is to create a `Grant.addToResource` for this use case; option C is leave it as-is. Opinions?

Reply:
`Grant.addToPrincipalAndResource` seems like the right solution though.

Will add to the identity if possible (which it would need), but won't add to the identity if the identity doesn't have a policy (like an `AccountRootPrincipal`).

Seems okay.

Reply:
Whelp, option A it is then. :) I'll update.

Reply:
Ok. Updated!